### PR TITLE
feat: add TF cloudbuilder blueprint

### PR DESCRIPTION
**Pull request by bharathkkb**
This adds a Terraform cloud-builder blueprint which creates a set of resources for automating CloudBuild runner image creation, management and scheduled updates following upstream Terraform releases.

### CODE REVIEW

**Comment by comment-bot-dev** [comment]
Thanks for the PR! 🚀
Unfortunately it looks like some of our CI checks failed. See the [Contributing Guide](../blob/master/CONTRIBUTING.md) for details.
- ⚠️check_whitespace
 Failed whitespace check. More details below. 
 ```bash 
Checking for trailing whitespace
./modules/tf_cloudbuild_builder/templates/workflow.yaml.tftpl:31:                    raise: $${e}                    
./modules/tf_cloudbuild_builder/cb.tf:21:  # substitutions available in the CB trigger 
./modules/tf_cloudbuild_builder/README.md:24:- Cloud Workflow for querying upstream Terraform releases, existing TF Builder images and triggering a Cloud Build build if an image does not exist for the latest Terraform release 
Error: Trailing whitespace found in the lines above.
Checking for missing newline at end of file
 ``` 
- ⚠️check_shell
 Failed shell check. More info on running shellcheck locally [here](https://www.shellcheck.net). 
 ```bash 
Running shellcheck
In ./examples/tf_cloudbuild_builder_simple/scripts/push-to-repo.sh line 32:
gcloud source repos clone ${CSR_NAME} ${tmp_dir} --project ${CSR_PROJECT_ID}
                          ^---------^ SC2086: Double quote to prevent globbing and word splitting.
                                      ^--------^ SC2086: Double quote to prevent globbing and word splitting.
                                                           ^---------------^ SC2086: Double quote to prevent globbing and word splitting.
In ./examples/tf_cloudbuild_builder_simple/scripts/push-to-repo.sh line 33:
cp ${DOCKERFILE_PATH} ${tmp_dir}
   ^----------------^ SC2086: Double quote to prevent globbing and word splitting.
                      ^--------^ SC2086: Double quote to prevent globbing and word splitting.
In ./examples/tf_cloudbuild_builder_simple/scripts/push-to-repo.sh line 34:
pushd ${tmp_dir}
      ^--------^ SC2086: Double quote to prevent globbing and word splitting.
For more information:
  https://www.shellcheck.net/wiki/SC2086 -- Double quote to prevent globbing ...
 ```

**Comment by comment-bot-dev** [comment]
@bharathkkb
Thanks for the PR! 🚀<br/>✅ Lint checks have passed.

**Comment by comment-bot-dev** [comment]
@bharathkkb
Thanks for the PR! 🚀<br/>✅ Lint checks have passed.

**Comment by comment-bot-dev** [comment]
@bharathkkb
Thanks for the PR! 🚀<br/>✅ Lint checks have passed.

**Comment by comment-bot-dev** [comment]
@bharathkkb
Thanks for the PR! 🚀<br/>✅ Lint checks have passed.

**Comment by comment-bot-dev** [comment]
@bharathkkb
Thanks for the PR! 🚀<br/>✅ Lint checks have passed.

File: examples/tf_cloudbuild_builder_simple/main.tf [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+  source = "../../modules/tf_cloudbuild_builder"
+
+  project_id          = module.enabled_google_apis.project_id
+  dockerfile_repo_uri = google_sourcerepo_repository.builder_dockerfile_repo.url
+  # allow logs bucket to be destroyed
+  cb_logs_bucket_force_destroy = true
+}
+
+# CSR for storing Dockerfile
+resource "google_sourcerepo_repository" "builder_dockerfile_repo" {
```
Comment (g-awmalik) [inline examples/tf_cloudbuild_builder_simple/main.tf:27]: Can this be created in the cloud_builder module?

File: modules/tf_cloudbuild_builder/cb.tf [hunk 2]
```diff
@@ -0,0 +1,114 @@
...
+  source_repo_url_split = local.is_source_repo ? split("/", var.dockerfile_repo_uri) : []
+  source_repo_project   = local.is_source_repo ? local.source_repo_url_split[4] : ""
+  source_repo_name      = local.is_source_repo ? local.source_repo_url_split[6] : ""
+}
+
+resource "google_cloudbuild_trigger" "build_trigger" {
+  project     = var.project_id
+  name        = var.trigger_name
+  description = "Builds a Terraform runner image. Managed by Terraform."
+  source_to_build {
```
Comment (g-awmalik) [inline modules/tf_cloudbuild_builder/cb.tf:41]: Does this blueprint support GCS style deployment? Something that was important to NetApp

**Review by g-awmalik — COMMENTED** [review]
Few comments. Overall looking great.

File: examples/tf_cloudbuild_builder_simple/main.tf (hunk 1 above)
Comment (bharathkkb) [reply examples/tf_cloudbuild_builder_simple/main.tf:27]: @g-awmalik For now I kept the source repo outside the module to allow 3rd party VCS

**Review by g-awmalik — APPROVED** [review]

**Pull request merged**